### DISCUSSION TITLE
PorousFlow mechanics with an empty pipe in a porous medium

### DISCUSSION THREAD

**Discussion by marinsiebert** [idaholab/moose#28251] [Q&A Modules: Porous Flow]
Hello everyone, 

I have a functioning input file for a TH-problem where a heat-source is placed in a porous medium. The heat-source is a fluid filled pipe, the thermal processes in the pipe are currently calculated by simple HeatConduction Kernels outside of the PorousFlow module. 
How would I proceed to incorporate mechanics in order to investigate consolidation of the porous medium surrounding the pipe?

**Comment by josebastiase** [comment]
Well, you just need to couple the mechanical part to your model

For a THM problem look at modules/porous_flow/examples/tutorial/04.i and for the consolidation problem here modules/porous_flow/test/tests/poro_elasticity/terzaghi_constM.i

**Reply by marinsiebert** [reply]
Thank you for your reply. For the porous medium I already have the blocks accordingly to the tutorials. The question is how to consider the mechanics for the pipe, as stress would also be taken and transferred by the pipe, and it would also be displaced when the underlying soil is consolidated. 
I probably phrased this a little bit unclear in the initial question.

**Reply by josebastiase** [reply]
With TransientMultiApps maybe? (https://mooseframework.inl.gov/syntax/MultiApps/index.html). With two input files one for your porous medium and one for your pipe. You need to transfer data from one to the other. But this is not fully coupled.

**Reply by GiudGiud** [reply]
Iterative coupling with multiapp s is "tight coupling" and it converges to the same answer as full coupling (single matrix approach)

**Reply by marinsiebert** [reply]
I've got the mechanics running so far, however the results seem a little counterintuitive.
The calculated stresses are positive, per convention this means they are expansive rather than compressive. Correspondingly, the resulting displacements are positive as well. 

![grafik](https://github.com/user-attachments/assets/4423ddd4-fbb2-4631-a11f-f3ad2faa45bb)



[test.txt](https://github.com/user-attachments/files/16450930/test.txt)


I guess this comes from the pore pressure being calculated by porepressure = airpressure (1 atm) - capillary pressure (here roughly 15kPa) 
This leads to a positive porepressure larger than the calculated stresses and thus expansion. 
However, porepressure is normally given by rho_w  * g * z and with the density of water smaller than of that soil should yield negative stresses. Am I missing something here?

**Reply by marinsiebert** [reply]
Giving this a push, as I am somewhat on a schedule. 
Looking at #23541 and the other threads cited by @WilkAndy a porepressure of 0 corresponds to atmospheric pressure and negative porepressures to an unsaturated medium - this does not seem to be the case here. I've created a simplified model of the problem described above using a single phase with porepressure as main variable and with this I am now able to achieve the desired results. This is fine for me. 
However, the question still remains if this is actually achievable with the NCG model?

**Reply by GiudGiud** [reply]
@cpgr
